test/cpp/test_aten_xla_tensor.cpp

```diff
@@ -5500,38 +5500,41 @@ TEST_F(AtenXlaTensorTest, TestIndexAdd) {
+  std::vector<double> alphas{-1.0, 0.0, 1.0, 2.0};
+
   for (torch::ScalarType scalar_type :
-       {torch::kFloat, torch::kByte, torch::kChar, torch::kShort, torch::kInt,
+       {torch::kByte, torch::kFloat, torch::kChar, torch::kShort, torch::kInt,
         torch::kLong}) {
     for (int dim = -rank; dim < rank; ++dim) {
-      ForEachDevice([&](const torch::Device& device) {
-        torch::Tensor base =
-            isFloatingType(scalar_type)
-                ? torch::rand({5, 3, 7}, torch::TensorOptions(scalar_type))
-                : torch::randint(100, {5, 3, 7},
-                                 torch::TensorOptions(scalar_type));
-        torch::Tensor index =
-            torch::randint(0, base.size(dim), {index_size},
-                           torch::TensorOptions(torch::kLong));
-        std::vector<int64_t> value_sizes(base.sizes().begin(),
-                                         base.sizes().end());
-        int canonical_dim = dim < 0 ? dim + rank : dim;
-        value_sizes[canonical_dim] = index_size;
-        torch::Tensor value =
-            isFloatingType(scalar_type)
-                ? torch::rand(value_sizes, torch::TensorOptions(scalar_type))
-                : torch::randint(100, value_sizes,
-                                 torch::TensorOptions(scalar_type));
-        torch::Tensor xla_base = CopyToDevice(base.clone(), device);
-        torch::Tensor result = base.index_add_(dim, index, value, alpha);
-        torch::Tensor xla_index = CopyToDevice(index, device);
-        torch::Tensor xla_value = CopyToDevice(value, device);
-        torch::Tensor xla_result =
-            xla_base.index_add_(dim, xla_index, xla_value, alpha);
-        AllClose(result, xla_result);
-        AllClose(base, xla_base);
-      });
+      for (double alpha : alphas) {
+        ForEachDevice([&](const torch::Device& device) {
+          torch::Tensor base =
+              isFloatingType(scalar_type)
+                  ? torch::rand({5, 3, 7}, torch::TensorOptions(scalar_type))
+                  : torch::randint(100, {5, 3, 7},
+                                   torch::TensorOptions(scalar_type));
+          torch::Tensor index =
+              torch::randint(0, base.size(dim), {index_size},
+                             torch::TensorOptions(torch::kLong));
+          std::vector<int64_t> value_sizes(base.sizes().begin(),
+                                           base.sizes().end());
+          int canonical_dim = dim < 0 ? dim + rank : dim;
+          value_sizes[canonical_dim] = index_size;
+          torch::Tensor value =
+              isFloatingType(scalar_type)
+                  ? torch::rand(value_sizes, torch::TensorOptions(scalar_type))
+                  : torch::randint(100, value_sizes,
+                                   torch::TensorOptions(scalar_type));
+          torch::Tensor xla_base = CopyToDevice(base.clone(), device);
+          torch::Tensor result = base.index_add_(dim, index, value, alpha);
```

Comment:
This (native torch implementation, bypassing lowering) fails when `scalar_type=kByte` with:
```
++ exception with description "value cannot be converted to type uint8_t without overflow" thrown in the test body.
```
cc @JackCaoG any suggestion?

Reply:
I am guessing it is because when `alpha = 2`, `value = 100` and `base = 100` it will overflow. Since we have alpha scaling, we can just make `value` with upper_bound of `50` instead of `100`.

Reply:
Awesome, and it's unsigned. I think the real culprit might be `alpha=-1.0` which will cause overflow. Let me try to limit the `randint` range as well as the `alphas`.